### DISCUSSION TITLE
Folders for the streamdeck (previous git)

### DISCUSSION THREAD

**Discussion by Enoratato** [streamdeck-linux-gui/streamdeck-linux-gui#7] [Ideas / Feature Requests]
Down below is my previous feature request. I wanted to ask if this fork will be implementing this feature at some point in the future. Thanks.

old

Hello, I wanted to ask if it's possible to generate folders near endlessly like the official version of StreamDeck app

For example, right-clicking on one of the buttons on the UI and creating a folder. In concept, similar to pages but without a limitation.

**Comment by coolapso** [comment]
If someone wants to work on it and make a pull request ill beg glad to review, test and accept it.

**Comment by Danie10** [comment]
Maybe I'm not understanding this request quite clearly? We have buttons belong to pages (could all be saved and opened from a named profile file). The "folders" on the original Windows one being "pages" on the Linux app? So we are actually asking for unlimited Pages?

**Reply by Enoratato** [reply]
Yeah unlimited pages to be more clear. Because in the stream deck app we only have a limit of 10 pages but we can make unlimited folders.

**Reply by My1** [reply]
well on windows it's a specific hierarchy and folders have their own pages too. so pages are linked together being essentially next to each other while folders are an upper/lower level.

**Comment by Danie10** [comment]
Got it. Yes, certainly more than 10 needed. Accompanying this feature, it would also be useful then to maybe have:
1. Naming of pages
2. Because space is a premium, keep the numbers as displayed, but have a tool tip pop-up on mouse hover, that displays the page's name.

Whether we want the naming/popup or not, the UI layout needs to consider how to display page numbers above 10 ie. scrolls sideways, wraps lines, etc?

**Reply by RemcoSchrijver** [reply]
Right now the limit is purely in the GUI, I think we should handle the page number as an id for lookup but we could attach a name attribute to a page possibly? I think the UI layout could work if we represent it as a graph? For my setup that would atleast work well because there are quite a lot of navigations to and from different pages.

**Comment by coolapso** [comment]
With the unlimited pages feature I assume this feature can be considered done? 

One can make from the pages whatever its required, and set the buttons to change to the designated page or is there something more?
